src/raftstore/store/worker/apply.rs

```diff
@@ -1386,65 +1510,71 @@ impl ApplyDelegate {
                 self.tag, state
             ),
         }
-        let mut exist_region = state.get_region().to_owned();
-        self.catch_up_log_for_merge(ctx, merge, &mut exist_region);
-        if *source_region != exist_region {
-            panic!(
-                "{} source_region {:?} not match exist region {:?}",
-                self.tag, source_region, exist_region
-            );
-        }
-        let mut region = self.region.clone();
-        // Use a max value so that pd can ensure overlapped region has a priority.
-        let version = cmp::max(
-            source_region.get_region_epoch().get_version(),
-            region.get_region_epoch().get_version(),
-        ) + 1;
-        region.mut_region_epoch().set_version(version);
-        if keys::enc_end_key(&region) == keys::enc_start_key(source_region) {
-            region.set_end_key(source_region.get_end_key().to_vec());
-        } else {
-            region.set_start_key(source_region.get_start_key().to_vec());
-        }
-        write_peer_state(&self.engine, ctx.wb(), &region, PeerState::Normal, None)
-            .and_then(|_| {
-                // TODO: maybe all information needs to be filled?
-                let mut merging_state = MergeState::new();
-                merging_state.set_target(self.region.clone());
-                write_peer_state(
-                    &self.engine,
-                    ctx.wb(),
-                    source_region,
-                    PeerState::Tombstone,
-                    Some(merging_state),
-                )
-            })
-            .unwrap_or_else(|e| {
-                panic!(
-                    "{} failed to save merge region {:?}: {:?}",
-                    self.tag, region, e
-                )
-            });
+        let exist_region = state.get_region().to_owned();
+        box self.catch_up_log_for_merge(ctx, merge, exist_region)
```

Comment:
box is not needed. and you can remove the box in the function signature as well.